packages/pwa-buildpack/src/WebpackTools/PWADevServer.js

```diff
@@ -179,47 +178,48 @@ be configured to have the same effect as 'id'.
             console.warn(secureHostWarning + helpText);
         }
 
-        const { graphqlPlayground } = config;
-        if (graphqlPlayground) {
-            const { queryDirs = [] } = config.graphqlPlayground;
+        if (config.graphqlPlayground) {
+            const noop = () => {};
             const endpoint = '/graphql';
 
-            const queryDirListings = await Promise.all(
-                queryDirs.map(async dir => {
-                    const files = await readdir(dir);
-                    return { dir, files };
-                })
-            );
+            const oldBefore = devServerConfig.before;
+            devServerConfig.before = (app, server) => {
+                let middleware;
+                let tabs;
+                oldBefore(app, server);
+                // this middleware has a bad habit of calling next() when it
+                // should not, so let's give it a noop next()
+                const { compiler } = server.middleware.context;
+                compiler.hooks.done.tap('PWADevServer', async stats => {
+                    const queryFilePaths = [];
+                    for (const filename of stats.compilation.fileDependencies) {
+                        if (filename.endsWith('.graphql')) {
+                            queryFilePaths.push(filename);
+                        }
+                    }
 
-            const queryDirContents = await Promise.all(
-                queryDirListings.map(({ dir, files }) =>
-                    Promise.all(
-                        files.map(async queryFile => {
-                            const fileAbsPath = resolve(dir, queryFile);
-                            const query = await readFile(fileAbsPath, 'utf8');
-                            const name = relative(process.cwd(), fileAbsPath);
+                    tabs = await Promise.all(
```

Comment:
Do we have to `await` here?  This is the last statement of this function and the `await` on L215 will take care of waiting for the `Promise.all` to finish, right?